zone/npc_scale_manager.cpp

```diff
@@ -30,21 +30,27 @@ void NpcScaleManager::ScaleNPC(
 	NPC *npc,
 	bool always_scale,
 	bool override_special_abilities
-)
-{
+) {
 	if (npc->IsSkipAutoScale() || npc->GetNPCTypeID() == 0) {
 		return;
 	}
 
-	int8 npc_type       = GetNPCScalingType(npc);
-	int  npc_level      = npc->GetLevel();
-	bool is_auto_scaled = IsAutoScaled(npc);
+	auto npc_type         = GetNPCScalingType(npc);
+	auto npc_level        = npc->GetLevel();
+	auto is_auto_scaled   = IsAutoScaled(npc);
+	auto zone_id          = npc->GetZoneID();
+	auto instance_version = npc->GetInstanceVersion();
 
-	global_npc_scale scale_data = GetGlobalScaleDataForTypeLevel(npc_type, npc_level);
+	global_npc_scale scale_data = GetGlobalScaleDataForTypeLevel(
+		npc_type,
+		npc_level,
+		zone_id,
```

Comment:
Same thing here, 0 lookups will fail if we pass in a specific zone. We'll need to look for specific first and then fallback to 0 if not found